### PR TITLE
[lldb] Fix progress reporting for SymbolLocatorDebugSymbols

### PR DESCRIPTION
**Pull request by JDevlieghere**
This fixes two issues related to the DebugSymbols symbol locator:

  1. Only the default symbol locator plugin reports progress. On Darwin, which uses the DebugSymbols framework we need to report the same progress form the corresponding SymbolLocator plugin.

  2. Forceful dSYM lookups, for example when using `add-dsym`, use a different code path that currently does not report progress, which is confusing. Here the progress event can be more specific and specify its downloading a symbol file rather than just locating it as we'll always shell out to dsymForUUID or its equivalent.

rdar://121629777

### CODE REVIEW

**Comment by chelcassanova** [comment]
LGTM, thanks for adding new reports!

File: lldb/source/Plugins/SymbolLocator/DebugSymbols/SymbolLocatorDebugSymbols.cpp [hunk 1]
```diff
@@ -776,6 +776,10 @@ std::optional<FileSpec> SymbolLocatorDebugSymbols::LocateExecutableSymbolFile(
       exec_fspec ? exec_fspec->GetFilename().AsCString("<NULL>") : "<NULL>",
       arch ? arch->GetArchitectureName() : "<NULL>", (const void *)uuid);
 
+  Progress progress(
+      "Locating external symbol file",
```
Comment (clayborg) [inline lldb/source/Plugins/SymbolLocator/DebugSymbols/SymbolLocatorDebugSymbols.cpp:780]: Are we looking for an executable here, or the dSYM file? Should be title be "Locating executable file"?

Comment (JDevlieghere) [reply lldb/source/Plugins/SymbolLocator/DebugSymbols/SymbolLocatorDebugSymbols.cpp:780]: We're looking for the symbol file (although dsymForUUID can also get you the symbol rich binary). This is the same message already emitted by `SymbolLocatorDefault::LocateExecutableSymbolFile`.

File: lldb/source/Plugins/SymbolLocator/DebugSymbols/SymbolLocatorDebugSymbols.cpp [hunk 2]
```diff
@@ -776,6 +776,10 @@ std::optional<FileSpec> SymbolLocatorDebugSymbols::LocateExecutableSymbolFile(
...
+      module_spec.GetFileSpec().GetFilename().AsCString("<Unknown>"));
```
Comment (adrian-prantl) [inline lldb/source/Plugins/SymbolLocator/DebugSymbols/SymbolLocatorDebugSymbols.cpp:781]: Aside: Looks like we're missing a AsString() method in ConstString so we can avoid the roundtrip through `char *`

File: lldb/source/Plugins/SymbolLocator/DebugSymbols/SymbolLocatorDebugSymbols.cpp [hunk 3]
```diff
@@ -1050,28 +1054,25 @@ bool SymbolLocatorDebugSymbols::DownloadObjectAndSymbolFile(
...
-                   dsymForUUID_exe_path.c_str(), copy_executable_arg,
-                   file_path_str.c_str());
-    LLDB_LOGF(log, "Calling %s with file %s to find dSYM: %s",
-              dsymForUUID_exe_path.c_str(), file_path_str.c_str(),
-              command.GetString().data());
-  } else {
-    return false;
-  }
+
+  StreamString command;
```
Comment (adrian-prantl) [inline lldb/source/Plugins/SymbolLocator/DebugSymbols/SymbolLocatorDebugSymbols.cpp:1065]: more unnecessary comments about unnecessary c string conversions :-)
```
    llvm::SmallString<64> buf;
    llvm::raw_svector_ostream(buf) << ...;
```

**Review by adrian-prantl — APPROVED** [review]

File: lldb/source/Plugins/SymbolLocator/DebugSymbols/SymbolLocatorDebugSymbols.cpp (hunk 2 above)
Comment (JDevlieghere) [reply lldb/source/Plugins/SymbolLocator/DebugSymbols/SymbolLocatorDebugSymbols.cpp:781]: https://github.com/llvm/llvm-project/pull/79649

**Pull request merged**